### ISSUE TITLE
"Other information" section not accurately labelled

### ISSUE THREAD

**Issue by emmastephenson** [CDCgov/prime-simplereport#3985]
## Area or Component
Patient Experience - Self registration

## Quick Summary
The form section for "other information" isn't a very accurate label. It's the name for the fieldset, but doesn't accurately label the questions.

## Accessibility Violation
Name: The element's name is missing or incorrect
https://axe.deque.com/rule-help/en/aria-name-missing-incorrect

## Action Requested
Change the label to be more accurate. See other examples on the form for examples (Demographics, Race, etc)

## Source Code (optional)
https://github.com/CDCgov/prime-simplereport/blob/c3e6b2dc330021e425e01657eaee613c51f9e11c/frontend/src/app/patients/Components/PersonForm.tsx

Note: This is likely going to be solved at the `PersonForm` level, meaning we also need to check the in-app `Add/Edit Person` page since the component is shared.

## Original Deque Issue
aria-name-missing-incorrect | The element's accessible name is missing or is not appropriate for the element's purpose. | "Other information" is the name for the fieldset, but doesn't accurately label the questions | Name: The element's name is missing or incorrect | https://axe.deque.com/rule-help/en/aria-name-missing-incorrect | critical | Yes | No | Interactive Elements | ["body > div#root:nth-of-type(1) > div.bg-base-lightest.minh-viewport:nth-of-type(1) > div.bg-base-lightest:nth-of-type(3) > div#registration-container.grid-container.maxw-tablet.padding-y-3:nth-of-type(1) > div.prime-formgroup:nth-of-type(4) > fieldset.usa-fieldset:nth-of-type(1)"] | The element's accessible name is missing or is not appropriate for the element's purpose. | <fieldset class="usa-fieldset"> | wcag2a,wcag412b | rnv7@cdc.gov | Patient Self-Registration - Form | https://stg.simplereport.gov/app/register/9C7BB |   | 2022-07-13T22:15:28.175Z
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --

**Comment by lauraykerr** [comment]
Would recommmend `Housing and work` for the last section

**Comment by lauraykerr** [comment]
@zdeveloper Including this in eng backlog as part of patient experience work.
